crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,152 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+pub struct OctezClientConfig {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientConfig {
+    pub fn new(endpoint: OctezNodeEndpoint) -> Self {
+        Self {
+            binary_path: None,
+            base_dir: None,
+            endpoint,
+            disable_unsafe_disclaimer: false,
+        }
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = endpoint;
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .take()
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
+        OctezClient {
```

Comment:
Just my opinion, `config.build` generating a client instead of a built config seems less natural to me compared with `Client::new(config_builder.build())`. I'm fine if others are okay with this implementation.

Reply:
i've followed the spec here but i agree with you. what about just renaming the `OctezClientConfig` to `OctezClientBuilder`? cc @zcabter

Reply:
Good idea, I'm in favour of `OctezClientBuilder` over `OctezClientConfig`